modules/virt-creating-single-pvc-template-storage-pool.adoc

```diff
@@ -0,0 +1,54 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virtual_disks/virt-configuring-local-storage-for-vms.adoc
+
+:_content-type: PROCEDURE
+[id="virt-creating-single-pvc-template-storage-pool_{context}"]
+= Creating a single `PVCTemplate` storage pool in a host path provisioner custom resource
+
+Specifying a single `pvcTemplate` (a standard Kubernetes `PersistentVolumeClaimSpec`)  allows you to specify a persistent volume claim (PVC) template, which is mounted on the node at the where you want to create hostpath provisioner (HPP) volumes. Using single `pvcTemplate`s can improve I/O performance. By configuring a cluster where each node has an additional disk to store the HPP, you effectively distribute many large volumes over multiple smaller virtual volumes.
+
+[NOTE]
+====
+You can use the Local Storage Operator to create provisioned volumes that can used by the HPP.
+====
+
+.Procedure
+
+. Create a YAML file for the HPP CSI driver provisioner custom resource specifying a single `pvcTemplate`. For example:
++
+[source,terminal]
+----
+$ touch hostpathprovisioner_cr_singlepvc.yaml
+----
+
+. Edit the file. For example:
++
+[source,yaml]
+----
+apiVersion: hostpathprovisioner.kubevirt.io/v1beta1
+kind: HostPathProvisioner
+metadata:
+ name: hostpath-provisioner
+spec:
+ imagePullPolicy: IfNotPresent
+ storagePools:
+   - name: local
+  pvcTemplate:
+       volumeMode: Block <1>
+       storageClassName: lso <2>
+       accessModes:
+       - ReadWriteOnce
+       resources:
+         requests:
+           storage: 5Gi <3>
+     path: "</var/myvolumes>"
+ workload:
+   nodeSelector:
+     kubernetes.io/os: linux
+----
+<1> `volumeMode` can be `Block` or `Filesystem`, but must match the provisioned volume format. Default is `Filesystem`. Note that the Local Storage Operator (LSO) writes only in `block` format. If the `volumeMode` is `block`, then the mounting pod creates an XFS filesystem on the block volume before mounting it.
+<2> If the `storageClassName` is omitted, the default storage class is used to create PVCs. If you omit `storageClassName`, ensure that the HPP storage class is not the default storage class.
+<3> In this example, the storage size is `5Gi`, which specifies that the volume must have at least five gigabytes available. Kubernetes binds volumes with available space greater than or equal to (>=) five gigabytes. Adjust the size according to the size available in your cluster. To ensure successful allocations, pick the smallest value possible, such as `1Mi`. If the storage class you are using as the source uses dynamically provisioned storage, pick an allocation size that matches the size of the request.
```

Comment:
So this blurp is only accurate if you have existing PVs that are very large compared to the PVC size you are requesting. Which is typically true with PVs created by LSO. This is why I mentioned this in the document you based this on. If you have dynamically created PV, then the PV will be the size you request, and if you request 1Mi, it is not going to be very useful. So in short, just mention this should be large enough to be useful but not so large that it cannot be satisfied by kubernetes.

Reply:
We condensed and reworded this.